streaming/src/main/scala/io/gearpump/streaming/appmaster/TaskManager.scala

```diff
@@ -98,6 +85,7 @@ private[appmaster] class TaskManager(
         val resourceRequestDetail = subDAGManager.executorFailed(executorId)
         executorManager ! StartExecutors(resourceRequestDetail.requests, resourceRequestDetail.jar)
       } else {
+        appMaster ! ShutdownApplication(appId)
```

Comment:
instead of telling AppMaster to shutdown, maybe it is better to tell AppMaster what is going wrong. It is AppMaster's responsibility to decide how to react to that error.